core/formatter.cpp

```diff
@@ -1724,16 +1734,245 @@ class FixIndentation {
+        UString key;
+        Fodder afterFodder;
+        Local::Bind *bind;
+        bool operator<(const ImportElem &elem) const {
+            return key < elem.key;
+        }
+    };
+
+    typedef std::vector<ImportElem> ImportElems;
+
+    Allocator &alloc;
+
+    public:
+    SortImports(Allocator &alloc) : alloc(alloc) { }
+
+    /// Get the value by which the imports should be sorted.
+    /// The imports are sorted by UTF-32 codepoints without case folding
+    UString sortingKey(Import *import)
+    {
+        return import->file->value;
+    }
+
+    /// Check if `local` expression is used for importing,
+    bool isGoodLocal(Local *local)
+    {
+        for (const auto &bind: local->binds) {
+            if (bind.body->type != AST_IMPORT) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    Local *goodLocalOrNull(AST *expr)
+    {
+        if (auto *local = dynamic_cast<Local*>(expr)) {
+            return isGoodLocal(local) ? local : nullptr;
+        } else {
+            return nullptr;
+        }
+    }
+
+    /** Split openFodder into what logically belongs to the previous AST node
+     * what belongs to the current AST node.
+     *
+     * Example:
+     * local x = import "xxx.jsonnet"; // Importing stuff here
+     *
+     * // blah blah
+     * {}
+     *
+     * In such case "// Importing  stuff here\n" part of the fodder belongs
+     * to the `local` and the rest to the {}.
+     */
+    std::pair<Fodder, Fodder> splitOpenFodder(AST *expr)
+    {
+        const Fodder &fodder = open_fodder(expr);
+        Fodder afterPrev, beforeThis;
+        bool inSecondPart = false;
+        for (const auto &fodderElem: fodder) {
+            if (inSecondPart) {
+                fodder_push_back(beforeThis, fodderElem);
+            } else {
+                afterPrev.push_back(fodderElem);
+            }
+            if (fodderElem.kind != FodderElement::Kind::INTERSTITIAL) {
+                inSecondPart = true;
+                if (fodderElem.blanks > 0) {
+                    afterPrev.back().blanks = 0;
+                    assert(beforeThis.empty());
+                    beforeThis.emplace_back(
+                        FodderElement::Kind::LINE_END,
+                        fodderElem.blanks,
+                        fodderElem.indent,
+                        std::vector<std::string>()
+                    );
+                }
+            }
+        }
+        return {afterPrev, beforeThis};
+    }
+
+    void sortGroup(ImportElems &imports)
+    {
+        // We don't want to change behaviour in such case:
+        // local foo = "b.jsonnet";
+        // local foo = "a.jsonnet";
+        // So we don't change the order when there are shadowed variables.
+        if (!duplicatedVariables(imports)) {
+            std::sort(imports.begin(), imports.end());
+        }
+    }
+
+    ImportElems extractImportElems(Local *local, Fodder after)
```

Comment:
Done, i.e. I have fixed it. I may add some more tests.

BTW I see some real value in it - in discoverability/UX. Otherwise the first-time user may assume that import sorting doesn't work or that it requires some additional setup and never change the style (how would they know that they need to use one-local-per-import?).